website/hw/hw6.rst

```diff
@@ -35,406 +43,480 @@ repository, and run some tests. These actions are typically referred to
+In the previous task we saw that GitHub Actions can build our code and run
 the tests, and alert us to any issues when doing so. However, our job
 was running specifically in an Ubuntu 16.04 environment. What if our
 code doesn’t compile in other environments? CI systems can also help us
 with this, as they often provide mechanisms to easily build our code in
 multiple environments. For example, we may want to build our code in
 multiple Ubuntu versions, or using different compilers.
 
-In Travis CI, we have the ability to specify build options that will
-automatically generate additional jobs. For example, add the following
-to your ``.travis.yml`` file:
+In GitHub Actions, we can do this by specifying a *matrix* of job
+configurations. For example, add this option to your ``build-and-test``
+job:
 
 ::
 
-   compiler:
-     - gcc
-     - clang
+    strategy:
+      matrix:
+        os: [ubuntu-16.04, ubuntu-18.04]
+        compiler: [gcc, clang]
 
-This tells Travis CI that we want to run the build with two compilers:
-``gcc`` and ``clang``. If you commit and push this updated file, this
-will produce a build with two jobs:
+Commit and push this change. You'll notice that your workflow now
+produces four jobs, corresponding to every combination of the ``os``
+and ``compiler`` values shown above.
 
-.. figure:: /cmsc22000/img/compilers-build.png
-   :alt: compilers-build
+However, the ``matrix`` option above doesn't actually affect the
+running environment or compiler used in the jobs (for example,
+if you click through the details of the Ubuntu 18.04 clang job,
+you'll see that Ubuntu 16.04 still appears in the "Set up job"
+step, under "Operating System", and that the build steps still use GCC).
 
-   compilers-build
+The ``matrix`` option simply defines variables we can use in our
+workflow file, and which get substituted for each individual job.
+In this case, the variables are ``${{ matrix.os }}`` and ``${{ matrix.compiler }}``.
+So, if we replace this::
 
-You can click on the individual jobs to see the full job log, showing
-the exact commands that were run in each job. You’ll actually notice
-that the clang job seems to still be running gcc. Don’t worry about this
-for now (we’ll revisit this in Task 3)
+    runs-on: ubuntu-16.04
 
--  [10 points] Take the URL of the build you just produced, and paste it
-   into Gradescope (under “Task 2: Multiple compilers”).
+With this::
 
-Travis CI actually provides two mechanisms to specify multiple jobs:
+    runs-on: ${{ matrix.os }}
 
-**Using a “matrix expansion” option**. There are certain options in
-Travis, such as the ``compiler`` option that will automatically result
-in generating multiple jobs. Another such option is the ``arch`` option
-(for specifying the processor architecture our build will run on). For
-example, if we specified the following options:
+Each run will use the appropriate operating system. Give it a try!
```

Comment:
May be helpful here to specify that they **should** replace the `runs-on` line and commit/push that change